src/main/java/bio/terra/cli/businessobject/User.java

```diff
@@ -130,9 +131,14 @@ public static void login() {
     // populate the current user object or build a new one
     User user = currentUser.orElseGet(() -> new User());
 
-    // do the login flow if the current user is undefined or has expired credentials
-    if (currentUser.isEmpty() || currentUser.get().requiresReauthentication()) {
-      user.doOauthLoginFlow();
+    try {
```

Comment:
I think we should make this an explicit option on the login command instead of doing this by default (e.g. `terra login --use-application-default`). I set app-default creds on my machine for other things, and it would be unexpected if the CLI pulled those. Also it would often not work, since they're usually not my pet SA, but an unrelated SA.

Then you can call login with this option in the notebook `post-startup.sh` script.